exchanges/coinbasepro/coinbasepro_test.go

```diff
@@ -75,13 +76,64 @@ func TestGetTrades(t *testing.T) {
 	}
 }
 
-func TestGetHistoricRates(t *testing.T) {
-	_, err := c.GetHistoricRates(testPair, 0, 0, 0)
+// TestGetHistoricRates_api_check using a time range of 5 minutes and a granularity of 1 minutes, it should return exactly 5 candles
+func TestGetHistoricRates_api_check(t *testing.T) {
+	e := expectedCandles(5, 300, 60)
+	if e != nil {
+		t.Error(e)
+	}
+	e = expectedCandles(2, 600, 300)
+	if e != nil {
+		t.Error(e)
+	}
+	e = expectedCandles(2, 1800, 900)
+	if e != nil {
+		t.Error(e)
+	}
+	e = expectedCandles(2, 7200, 3600)
+	if e != nil {
+		t.Error(e)
+	}
+	e = expectedCandles(2, 43200, 21600)
+	if e != nil {
+		t.Error(e)
+	}
+	e = expectedCandles(2, 172800, 86400)
+	if e != nil {
+		t.Error(e)
+	}
+}
+
+func expectedCandles(expectedCandles int, timeRange time.Duration, candleGranularity int64) error {
+	iso8601format := "2006-01-02T15:04:05"
+	end := time.Now().UTC().Add(-time.Second * timeRange) // the latest candle may not yet be ready, so skipping to the previous one
+	start := end.Add(-time.Second * timeRange)
+	resp, err := c.GetHistoricRates(testPair, start.Format(iso8601format), end.Format(iso8601format), candleGranularity)
```

Comment:
I think we might be able to use time.RFC3339 const instead of the iso8601format.

Reply:
thanks for that, didn't know it. Fixed